clients/vault/src/oracle/agent.rs

```diff
@@ -57,82 +84,76 @@ async fn handle_message(
 	Ok(())
 }
 
-/// Start the connection to the Stellar Node.
-/// Returns an `OracleAgent` that will handle incoming messages from Stellar Node,
-/// and to send messages to Stellar Node
-pub async fn start_oracle_agent(
+pub async fn listen_for_stellar_messages(
 	config: StellarOverlayConfig,
-	secret_key: &str,
+	collector: Arc<RwLock<ScpMessageCollector>>,
+	secret_key_as_string: String,
 	shutdown_sender: ShutdownSender,
-) -> Result<OracleAgent, Error> {
-	let is_public_network = config.is_public_network();
+) -> Result<(),service::Error<crate::Error>> {
+	tracing::info!("listen_for_stellar_messages(): Starting connection to Stellar overlay network...");
 
-	tracing::info!("start_oracle_agent(): Starting connection to Stellar overlay network...");
+	let mut overlay_conn = connect_to_stellar_overlay_network(config.clone(), secret_key_as_string)
+		.await.map_err(|e|{
+			tracing::error!("listen_for_stellar_messages(): Failed to connect to Stellar overlay network: {e:?}");
+			service::Error::StartOracleAgentError
+		})?;
 
-	let mut overlay_conn = connect_to_stellar_overlay_network(config.clone(), secret_key).await?;
 	// use StellarOverlayConnection's sender to send message to Stellar
 	let sender = overlay_conn.sender();
 
-	let collector = Arc::new(RwLock::new(ScpMessageCollector::new(
-		config.is_public_network(),
-		config.stellar_history_archive_urls(),
-	)));
-	let collector_clone = collector.clone();
-
-	let shutdown_sender_clone = shutdown_sender.clone();
-	// disconnect signal sender tells the StellarOverlayConnection to close its TcpStream to Stellar
-	// Node
-	let (disconnect_signal_sender, mut disconnect_signal_receiver) = mpsc::channel::<()>(2);
-
-	let sender_clone = overlay_conn.sender();
-	tokio::spawn(async move {
-		loop {
-			tokio::select! {
```

Comment:
Removed tokio::select, since this task will close based on the `run_and_monitor_task()`.